### ISSUE TITLE
Provide a "Window" wall type to match the "Door" wall type (complete with window icon)

### ISSUE THREAD

**Issue by aaclayton** [foundryvtt/foundryvtt#5025]
Originally in GitLab by @Albinosheep

### Feature Summary
If you create a window, which can be opened, you have to use the door icon from the wall tool. This can be irritating for the players.

### User Experience
Allow the gm native to switch icons in the wall tool: door icon and window icon.
This should be visible in the locked state too.

### Priority/Importance
Medium priority, but I think it would be easy to add.

**Comment by aaclayton** [comment]
Comment from @Fridan99

> It would be nice if windows could be configured on the maps. I think that adding some option inside the doors to change the icon (door or window) would be enough.  

### User Experience

> I think the experience would be enhanced. Foundry v9 and we still don't have windows? 🤣 

The functionality would be exactly like a door setup, but it would be nice to be able to differentiate a door from a window. 

### Priority/Importance

> I don't think it is either more important or less important. It probably doesn't require much work (from my lack of knowledge) to be able to switch the icon between door/window, since the functionality of a window will be exactly the same as the door, and it can give enrichment to the experience. 

That is why I put this suggestion, in case it is taken into account, since it is probably something “simple” to apply.

Thanks! And good work with this marvelous vtt.

**Comment by apoapostolov** [comment]
A particularly important behavior, should Foundry introduce support for windows, is that they are truly transparent when close enough and reflective and hard to see through from a further distance. 

This would probably be out of the scope of a quick window icon, but what would really enhance the experience is walls that become transparent only to actors who are within the distance range of one of its two endpoints.

**Comment by Fridan99** [comment]
> Originally in GitLab by @Albinosheep
> 
> Thank you for your feedback! :-) I miss an **icon ** which the players can identify as a window. Atm the icon only can show a door-icon, not a window-icon. I dont miss any sight-functions. ![image](https://camo.githubusercontent.com/5a6d910780f3168aa5e9475df358b541539965ee7773692546de0377c3cda11e/68747470733a2f2f6769746c61622e636f6d2f666f756e6472796e65742f666f756e6472797674742f75706c6f6164732f63353831346436366366366331306134336237656463383639663630666236362f646f6f722d77696e646f772d69636f6e2e706e67)

I miss a window **icon ** too, nothing more and nothing less.

**Comment by p4535992** [comment]
This is done with the v11, i think we can close it

**Comment by dev7355608** [comment]
Should this be closed? There are no window icons yet, right?

**Comment by aaclayton** [comment]
It's correct that there is not a specialized icon. I suppose I can reopen this and reframe it as adding icon support.

**Comment by aaclayton** [comment]
Comment from @esheyw

In order to model a window with shutters/curtains properly, you have to overlay two walls: one that lets light through (either always or as one of the new v11 windows), and a regular, light-stopping door directly on top of it. This works, but looks sort of silly, since it's not a door, and even with the 'door' open, the 'window' still prevents pathing.

If there's some way of working a feature like this into the core workflow like 'if door, allow different light/sound/movement blocking values per state', that'd be great, but probably way too much work. The two overlaid walls works well enough, even if it's slightly annoying (some method for 'cycling the stack' of walls would be great though - currently you have to just shift one node of the 'upper' wall to get at the lower)

But if we can't get that, could we at least get settable icons per door segment? There's a module that handles this: [Designer Doors](https://github.com/Exitalterego/designerdoors) but I think this is a simple/broad enough use case it should be doable in core in some fashion.

**Comment by caewok** [comment]
Regarding better modeling of things like windows with shutters/curtains: I think this is eminently doable with tweaks to the current API and UI. 

Currently, walls track door state using two booleans: `isDoor`and `isOpen`. Once the door is open, the wall is treated as not being there at all. If we add  a "door open state" object with values for `light`, `sight`, `sound`, and `move`, the wall representing the door could transition from one type of wall in the closed state to a different type of wall in the open state. 

To take the example from @esheyw of a window that can have open or closed shutters, you might have:
```js
wall.document = {
  // First set of properties are the normal wall properties for a window blocked by a shutter.
  light: 20,
  move: 20,
  sight: 20,
  sound: 20,

  // openState allows light and sight through but nothing else. (It is a well-insulated window! No sounds!)
  openState: {
   light: 0,
   move: 20,
   sight: 0,
   sound: 20
  }
}
```

I don't foresee much compatibility issues with this approach because `isDoor` and `isOpen` remain the same. Some modules would need to be updated to recognize the new open door states, but in general those likely would fall back on treating an open door as if it were completely open to all properties, instead of just selected properties.

**Comment by Albinosheep** [comment]
I’d like to give this feature request a gentle bump from a current UX perspective.

Foundry now has Window Walls / Proximity Walls, which is great for modeling visibility through windows. However, the original UX problem still seems relevant: there is no clear core workflow for a wall segment that behaves like an interactable door/shutter/window while being visually represented to players as a window rather than as a door.

The most useful core-level version of this feature would probably be:

- an icon selector for door-like wall segments, including at least “door” and “window”
- the chosen icon should be visible in closed and locked states where appropriate
- ideally, window-like segments could still use door states such as open, closed, and locked
- optionally, open/closed states could have separate movement, sight, light, and sound restrictions

Modules such as Designer Doors can cover part of this today, but this feels like a common enough mapping use case that it would be valuable in core. It would improve player clarity without requiring a module or awkward workarounds such as overlapping multiple wall segments.

The main request is not necessarily new vision behavior, but clearer UX and visual semantics: players should be able to distinguish “this is a window” from “this is a door” at a glance.

**Comment by Fridan99** [comment]
> I’d like to give this feature request a gentle bump from a current UX perspective.
> 
> Foundry now has Window Walls / Proximity Walls, which is great for modeling visibility through windows. However, the original UX problem still seems relevant: there is no clear core workflow for a wall segment that behaves like an interactable door/shutter/window while being visually represented to players as a window rather than as a door.
> 
> The most useful core-level version of this feature would probably be:
> 
> * an icon selector for door-like wall segments, including at least “door” and “window”
> * the chosen icon should be visible in closed and locked states where appropriate
> * ideally, window-like segments could still use door states such as open, closed, and locked
> * optionally, open/closed states could have separate movement, sight, light, and sound restrictions
> 
> Modules such as Designer Doors can cover part of this today, but this feels like a common enough mapping use case that it would be valuable in core. It would improve player clarity without requiring a module or awkward workarounds such as overlapping multiple wall segments.
> 
> The main request is not necessarily new vision behavior, but clearer UX and visual semantics: players should be able to distinguish “this is a window” from “this is a door” at a glance.


I agree with you; we’re already on v14, and there still aren’t any visual icons for windows, even though there are window features. It’s a bit “nonsensical.” 

In theory, changing the icon from a door to a window shouldn’t be a difficult task, but I don’t know why it hasn’t been implemented yet. 

I think it’s very useful visually—I’d even say essential. After all, players act or interpret based on what the GM describes and what they can see in the environment, and seeing a door icon on a window can lead to confusion.

I think this is something that should be fixed to enhance the visual experience. 

🙂
